src/cursor/abstract_cursor.ts

```diff
@@ -489,7 +493,7 @@ export abstract class AbstractCursor<
         await this.fetchBatch();
       } while (!this.isDead || (this.documents?.length ?? 0) !== 0);
     } finally {
-      if (this.cursorOptions.timeoutMode === CursorTimeoutMode.ITERATION && this.cursorId != null) {
+      if (shouldRefresh) {
```

Comment:
Since this line is after fetchBatch the value at cursorId could have changed right? so is the stored condition still the same one we want to refresh on?

Reply:
Yes because the only time that `cursorId` is null is when the cursor is uninitialized, but to guard against any changes we make to that logic, we can check for existence of `timeoutContext` instead, because they're effectively testing the same thing.

Reply:
Let me know what you find. 

It sounds correct; my thinking is that if we've hit an error we should still have a null id because the operation didn't give us one, maybe we still set it to 0 somewhere or this logic runs in either case.

Reply:
The only two times we set the cursorId to null are in
* `AbstractCursor#constructor()`
* `AbstractCursor.rewind()`

Anywhere else, we are setting the `cursorId` from the server response (which is never null, provided that nothing went wrong server-side), so regardless of where we check the `cursorId`, in `AbstractCursor.next()/.tryNext()/.hasNext()`, it will still yield the correct behaviour of only calling `timeoutContext.refresh()/.clear()` when the cursor is already initialized and in iteration mode.

If it's initially null, then we'd still want to clear the clear the timeout if the cursor gets killed (id set to zero).